pandas/core/frame.py

```diff
@@ -3999,7 +4000,12 @@ def _combine_const(self, other, func, errors='raise', try_cast=True):
         new_data = self._data.eval(func=func, other=other,
                                    errors=errors,
                                    try_cast=try_cast)
-        return self._constructor(new_data)
+        result = self._constructor(new_data)
+        if is_number(other):
+            bool_cols = self.dtypes.apply(is_bool_dtype)
+            index = bool_cols[bool_cols].index
+            result[index] = result[index].astype(type(other))
```

Comment:
so on windows ``type(other)`` if ``other``is an int would yield ``np.int32``, while on all other systems this would be ``np.int64``.

is there a reason we don't just cast back to ``bool`` for these columns?

Reply:
Hmmm shouldn't these cols be coerced to the same `dtype` as `other`? Ie copy the behavior from python:

```
>>> type(1.0 * True)
<class 'float'>
>>> 1.0 * True
1.0
```